shared/utils/checkStatus.ts

```
@@ -0,0 +1,51 @@
+export const checkStatus = (request: Types.IRequestDataWithEvents | null) => {
+  const balance = BigInt(request?.balance?.balance ?? 0);
+  const expectedAmount = BigInt(request?.expectedAmount ?? 0);
+  const today = new Date();
+  let dueDate: Date | null = null;
+  const isPaid = balance >= expectedAmount ? "Paid" : "Partially Paid";
+
+  try {
+    const rawDueDate = request?.contentData?.paymentTerms?.dueDate;
+    if (rawDueDate) {
+      dueDate = new Date(rawDueDate);
+      if (isNaN(dueDate.getTime())) {
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)